daemon/image_delete.go

```diff
@@ -13,164 +11,261 @@ import (
 	"github.com/docker/docker/utils"
 )
 
-// FIXME: remove ImageDelete's dependency on Daemon, then move to graph/
-func (daemon *Daemon) ImageDelete(name string, force, noprune bool) ([]types.ImageDelete, error) {
-	list := []types.ImageDelete{}
-	if err := daemon.imgDeleteHelper(name, &list, true, force, noprune); err != nil {
+// ImageDelete deletes the image referenced by the given imageRef from this
+// daemon. Behavior is as follows:
+// 	- A **hard dependent** is defined as:
+// 	  - a pull or build using this image.
+// 	  - any descendent image.
+// 	  - any *running* container using this image.
+// 	- A **soft dependent** is defined as:
+// 	  - any *stopped* container using this image.
+// 	  - any tag references to this image.
+// 	- If a tag/digest reference is specified, that tag/digest is *always*
+// 	  removed regardless of whether or not the image it references has any hard
+// 	  or soft dependents.
```

Comment:
According the current `rmi` behavior, if the image has only one tag, `docker rmi` means deleting the tag as well as the layers of the image.